playwright-tests/tests/system-updates/update-history.spec.js

```diff
@@ -0,0 +1,64 @@
+import { expect } from "@playwright/test";
+import { test } from "../../util/test.js";
+import {
+  compareInstanceWeb4WithTreasuryFactory,
+  redirectWeb4,
+} from "../../util/web4.js";
+
+test.afterEach(async ({ page }) => {
+  await page.unrouteAll({ behavior: "ignoreErrors" });
+});
+
+test("should show update history", async ({ page, instanceAccount }) => {
+  const contractId = instanceAccount;
+  const isUpToDate = await compareInstanceWeb4WithTreasuryFactory(
+    instanceAccount
+  );
+  await redirectWeb4({
+    contractId,
+    page,
+    modifiedWidgets: {
+      "widgets.treasury-factory.near/widget/pages.settings.system-updates.UpdateRegistry": `
+    return [
+      {
```

Comment:
this is amazing, so we can pass the widget content directly, this mean we can test for each type of update, without actually adding it to the frontend code.

Reply:
yes!